server/rpc_server.go

```diff
@@ -255,6 +255,13 @@ func MonitorRPCRequest() func(ctx context.Context, methodName string, err error)
 	start := time.Now()
 	return func(ctx context.Context, methodName string, err error) {
 		m := rpcEndpointMetrics["rpc."+methodName]
+		if m==nil{
+			registeredRPCEndpoints := make([]string, 0, len(rpcEndpointMetrics)) //for a better error-message retrieve all correctly registered rpcEndpoints
+			for registeredRPCEndpoint := range rpcEndpointMetrics {
+				registeredRPCEndpoints = append(registeredRPCEndpoints, registeredRPCEndpoint[4:len(registeredRPCEndpoint)]) //to hide internals slice away the "rpc."-part
+			}
+			fmt.Printf("rpc server panic: the methodName '%v' listed in the second parameter of MonitorRPCRequest() was not registered with the server.\nRegistered methodNames are: %v. Please check for spelling mistakes.\n\n", methodName, registeredRPCEndpoints)
```

Comment:
Just noticed something:
The recover part itself calls `m.PanicCounter` which will crash, too.
That's unpleasant.

A potential new panic inside the recover-block...